Gems/Atom/RHI/Vulkan/Code/Source/RHI/BufferMemory.cpp

```diff
@@ -39,14 +74,54 @@ namespace AZ
             return result;
         }
 
+        bool BufferMemory::IsVmaAllocated() const
+        {
+            return m_vmaAllocation != VK_NULL_HANDLE;
+        }
+
+        const VmaAllocationInfo& BufferMemory::GetVmaAllocationInfo() const
+        {
+            AZ_Assert(IsVmaAllocated(), "Buffer memory was not allocated with VMA");
+            return m_vmaAllocationInfo;
+        }
+
         CpuVirtualAddress BufferMemory::Map(size_t offset, size_t size, RHI::HostMemoryAccess hostAccess)
         {
-            return m_memoryView.GetMemory()->Map(m_memoryView.GetOffset() + offset, size, hostAccess);
+            Device& device = static_cast<Device&>(GetDevice());
```

Comment:
What if the memory came from m_memoryView (i.e no vma)? Can we take out m_memoryView completely?